js.go

```diff
@@ -3121,6 +3121,7 @@ func (sub *Subscription) Fetch(batch int, opts ...PullOpt) ([]*Msg, error) {
 			if err := nc.PublishRequest(nms, rply, req); err != nil {
 				return err
 			}
+			connStatusChanged := nc.StatusChanged()
 			if o.hb > 0 {
```

Comment:
I think this should be triggered regardless of whether `o.hb > 0`. We want to be able to bail early even without heartbeat set.